timely_beliefs/beliefs/classes.py

```diff
@@ -419,6 +430,28 @@ def search_session(  # noqa: C901  # todo: remove after removing deprecated argu
             source_cls = sources[0].__class__
             q = q.join(source_cls).filter(cls.source_id.in_([s.id for s in sources]))
 
+        # Apply most recent beliefs filter
+        if most_recent_only:
+            subq = (
+                session.query(
+                    cls.event_start,
+                    cls.source_id,
+                    func.min(cls.belief_horizon).label("most_recent_belief_horizon"),
+                )
+                .group_by(cls.event_start, cls.source_id)
+                .order_by(func.min(cls.belief_horizon).desc())
+                .filter(cls.sensor_id == sensor.id)
```

Comment:
Would it be faster (and still work) if we first `filter()` and then `group_by()`? It could be that sqlachemy optimises this stuff but maybe not.

Would it work without this filter the same way? The main query has a few filters which you connect to after joining the subquery, I'm curious why this filter is actually in the subquery itself.